app/factories/shopify/variant_factory.rb

```diff
@@ -0,0 +1,31 @@
+module Shopify
+  class VariantFactory
+    def initialize(spree_variant, shopify_variant)
+      @spree_variant = spree_variant
+      @shopify_variant = shopify_variant
+    end
+
+    def perform
+      shopify_variant.weight = spree_variant.weight
+      shopify_variant.weight_unit = 'oz'
+      shopify_variant.price = spree_variant.price
+      shopify_variant.sku = spree_variant.sku
+      shopify_variant.updated_at = spree_variant.updated_at
+      generate_options!
+
+      shopify_variant
+    end
+
+    private
+
+    attr_accessor :spree_variant, :shopify_variant
+
+    def generate_options!
+      shopify_variant.option1 = spree_variant.sku
+
+      spree_variant.option_values.each_with_index do |option, index|
+        shopify_variant.send("option#{index + 2}=", option.name)
```

Comment:
Sorry for the comment I wrote, re-reading it and talking with Braiden made me realize that it was bad manner and it didn't leave place for discussion. I would have probably re-read it and made my point a little bit more explanatory. 

What I should have said is the following: 

I actually think that self-documenting code is kind of a myth. Something can't be clear for the majority of people. There will always be something that will make you have to dig anyway.

I just thought that we had that conversation many times in the past and we had decided to write comments when cases were nebulous and that's what I meant by writing "I'll write comments."

But at the same time, I remember having a conversation, I think in Frontendo, where we said that comments should be kept at a minimum.

Maybe I am just bad at finding write variables / method names to propagate the thinking behind the value of that said code.

But I would agree that what you have suggested below: `assign_variant_uniqueness_constraint`is speaking by itself and is far better than having a comment written. I jumped the gun too quickly before actually thinking about it.

My apologies for any confusions and I'll adjust for later comments! 💃